From a51f3cfe2b1704f7b187809423cb19055fd3eea5 Mon Sep 17 00:00:00 2001
From: Han Xu <b45815@freescale.com>
Date: Fri, 5 Jun 2015 15:09:18 -0500
Subject: [PATCH] MLK-11061: ARM:dts: enable ddr quad mode for QSPI on 6UL
 ARM2 and 7D ARM2

Enable DDR Quad mode for QSPI on 6UL ARM2 and 7D ARM2 board to improve
the performance.

Signed-off-by: Han Xu <b45815@freescale.com>
---
 arch/arm/boot/dts/imx6ul-14x14-evk.dts             | 1 +
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-qspi.dts | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/arch/arm/boot/dts/imx6ul-14x14-evk.dts b/arch/arm/boot/dts/imx6ul-14x14-evk.dts
index 8bd14d4..138c4b5 100644
--- a/arch/arm/boot/dts/imx6ul-14x14-evk.dts
+++ b/arch/arm/boot/dts/imx6ul-14x14-evk.dts
@@ -357,6 +357,7 @@
 		#size-cells = <1>;
 		compatible = "micron,n25q256a";
 		spi-max-frequency = <29000000>;
+		spi-nor,ddr-quad-read-dummy = <6>;
 		reg = <0>;
 	};
 };
diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-qspi.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-qspi.dts
index 4f2eb33..ede6df2 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-qspi.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-qspi.dts
@@ -50,6 +50,7 @@
 		#size-cells = <1>;
 		compatible = "micron,n25q256a";
 		spi-max-frequency = <29000000>;
+		spi-nor,ddr-quad-read-dummy = <6>;
 		reg = <0>;
 	};
 
@@ -58,6 +59,7 @@
 		#size-cells = <1>;
 		compatible = "micron,n25q256a";
 		spi-max-frequency = <29000000>;
+		spi-nor,ddr-quad-read-dummy = <6>;
 		reg = <1>;
 	};
 
@@ -66,6 +68,7 @@
 		#size-cells = <1>;
 		compatible = "micron,n25q256a";
 		spi-max-frequency = <29000000>;
+		spi-nor,ddr-quad-read-dummy = <6>;
 		reg = <2>;
 	};
 
@@ -74,6 +77,7 @@
 		#size-cells = <1>;
 		compatible = "micron,n25q256a";
 		spi-max-frequency = <29000000>;
+		spi-nor,ddr-quad-read-dummy = <6>;
 		reg = <3>;
 	};
 };
-- 
1.8.0

